pkg/handler/common/machine.go

```diff
@@ -461,6 +461,20 @@ func PatchMachineDeployment(ctx context.Context, userInfoGetter provider.UserInf
 	if err != nil {
 		return nil, fmt.Errorf("cannot output existing node deployment: %w", err)
 	}
+	var unmarshalPatched *apiv1.NodeDeployment
+	if err := json.Unmarshal(patch, &unmarshalPatched); err != nil {
+		return nil, fmt.Errorf("cannot decode patched nodedeployment: %w ===", err)
+	}
+
+	isOsCahnged := isOsCahnged(unmarshalPatched.Spec.Template.OperatingSystem)
```

Comment:
Maybe we can rename this to something like `selectedOperatingSystems`? `isOsCahnged` is a bit ambigous since we are checking for the existence of OS in the payload.